### PR TITLE
Fix #56 Add special handling for Planck REPLS

### LINKED ISSUE #56: Add special handling for planck REPLs
Basically we need to extend what was done in #44 to cover planck as well, so all the `inf-clojure` functionality would work with planck properly.

This should be pretty easy and I would have done it myself if I knew the first thing about planck. :-)

### PR DESCRIPTION
**Pull request by slipset**
Added support for the Planck REPL

Before submitting the PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines][1]
- [ ] The new code is not generating bytecode or `M-x checkdoc` warnings
- [x] You've updated the changelog (if adding/changing user-visible functionality)
- [x] You've updated the readme (if adding/changing user-visible functionality)

Thanks!

[1]: https://github.com/clojure-emacs/inf-clojure/blob/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: inf-clojure.el [hunk 1]
```diff
@@ -236,7 +237,15 @@ Clojure to load that file."
 (define-obsolete-variable-alias 'inf-clojure-load-command 'inf-clojure-load-form "2.0.0")
 
 (defcustom inf-clojure-load-form-lumo "(clojure.core/load-file \"%s\")"
-  "Format-string for building a Clojure expression to load a file.
+  "Planck format-string for building a Clojure expression to load a file.
```
Comment (arichiardi) [inline inf-clojure.el]: Maybe little typo here, shouldn't it be for Lumo? I have probably forgot to update this one myself :smile:

File: inf-clojure.el [hunk 2]
```diff
@@ -272,7 +282,7 @@ This should usually be a combination of `inf-clojure-prompt' and
   :package-version '(inf-clojure . "2.0.0"))
 
 (defvar inf-clojure-buffer nil
-  "The current inf-clojure process buffer.
+  "The current ‘inf-clojure’ process buffer.
```
Comment (arichiardi) [inline inf-clojure.el]: Oh just out of curiosity, why those forward ticks?

File: inf-clojure.el [hunk 3]
```diff
@@ -649,12 +680,19 @@ If you are using REPL types, it will pickup the most approapriate
   :type 'string
   :package-version '(inf-clojure . "2.0.0"))
 
+(defcustom inf-clojure-completion-form-planck
+  "(doall (map str (planck.repl/get-completions \"%s\")))\n"
```
Comment (arichiardi) [inline inf-clojure.el]: this `(doall (map str ...` was necessary because lumo returns `#js`. Does planck return it as well?

File: inf-clojure.el [hunk 4]
```diff
@@ -685,6 +730,20 @@ If you are using REPL types, it will pickup the most approapriate
   :type 'string
   :package-version '(inf-clojure . "2.0.0"))
 
+(defcustom inf-clojure-set-ns-form-planck
+  "(in-ns '%s)\n"
```
Comment (arichiardi) [inline inf-clojure.el]: no namespace here?

Comment (slipset) [inline inf-clojure.el]: Seems like `in-ns` lives in top-level
```Clojure
cljs.user=> (doc in-ns)
-------------------------
in-ns
([name])
REPL Special Function
  Sets *cljs-ns* to the namespace named by the symbol, creating it if needed.
nil
cljs.user=> (doc apropos)
-------------------------
planck.repl/apropos
([str-or-pattern])
Macro
  Given a regular expression or stringable thing, return a seq of all
  public definitions in all currently-loaded namespaces that match the
  str-or-pattern.
```

File: inf-clojure.el (hunk 3 above)
Comment (slipset) [inline inf-clojure.el]: 8929305

File: inf-clojure.el (hunk 2 above)
Comment (slipset) [inline inf-clojure.el]: checkdoc suggested it. removed
8929305

File: inf-clojure.el (hunk 1 above)
Comment (slipset) [inline inf-clojure.el]: 8929305

**Comment by slipset** [comment]
Thanks for the review @arichiardi!

**Comment by bbatsov** [comment]
👍

File: inf-clojure.el (hunk 4 above)
Comment (mfikes) [inline inf-clojure.el]: Right, Planck's `in-ns` is a REPL special (just like in the ClojureScript REPL). It has not namespace.

**Pull request merged**